src/perf/bin/drvmain.cpp

```diff
@@ -490,18 +502,136 @@ QuicPerfCtlEvtIoCanceled(
     WdfRequestComplete(Request, Status);
 }
 
+NTSTATUS
+QuicPerfCtlSetSecurityConfig(
+    _Inout_ QUIC_DRIVER_CLIENT* Client,
+    _In_ const QUIC_CERTIFICATE_HASH* CertHash
+)
+{
+    Client->SelfSignedConfiguration.SelfSignedSecurityHash = *CertHash;
+    Client->SelfSignedValid = true;
+    return QUIC_STATUS_SUCCESS;
+}
+
 size_t QUIC_IOCTL_BUFFER_SIZES[] =
 {
     0,
     sizeof(QUIC_CERTIFICATE_HASH),
-    0,
+    SIZE_MAX,
     0
 };
 
+typedef union {
+    struct {
+        int Length;
+        char Data;
+    };
+    QUIC_CERTIFICATE_HASH CertHash;
+} QUIC_IOCTL_PARAMS;
+
 static_assert(
     QUIC_PERF_MAX_IOCTL_FUNC_CODE + 1 == (sizeof(QUIC_IOCTL_BUFFER_SIZES) / sizeof(size_t)),
     "QUIC_IOCTL_BUFFER_SIZES must be kept in sync with the IOTCLs");
 
+QUIC_THREAD_CALLBACK(WaitThreadCb, Context)
```

Comment:
I don't understand what's going on in this thread from it's name. Can you improve it a bit?